match2/wikis/rocketleague/legacy_match_list.lua

```diff
@@ -0,0 +1,238 @@
+		if args.walkover then
+			if tonumber(args.walkover) == index then
+				score = 'W'
+			elseif Table.includes(ALLOWED_STATUSES, args['games' .. index]) then
+				score = args['games' .. index]
+			else
+				score = 'L'
+			end
+		else
+			score = args['games' .. index] or '0'
+		end
+		if player ~= 'TBD' then
+			args['opponent' .. index] = MatchSubobjects.luaGetOpponent(frame, {
+					type = 'solo',
+					match2players = '[' .. json.stringify({
+						name = playerLink,
+						displayname = player,
+						flag = args['p' .. index .. 'flag'],
+					}) .. ']',
+					template = args['p' .. index .. 'team'],
+					score = score,
+				})
+		end
+		args['opponent' .. index .. 'literal'] = args['team' .. index .. 'literal']
+
+		--atm we ignore the old teamXstanding parameters, because
+		--they are not supported in the new system
+
+		--empty all the stuff we set into this opponent
+		args['player' .. index] = nil
+		args['team' .. index] = nil
+		args['p' .. index .. 'flag'] = nil
+		args['p' .. index .. 'team'] = nil
+		args['team' .. index .. 'literal'] = nil
+		args['games' .. index] = nil
+		args.walkover = nil
+	end
+
+	--process maps
+	args, details = LegacyMatchList.processMaps(args, details)
+
+	--process other stuff from details
+	args = LegacyMatchList.copyDetailsToArgs(args, details)
+
+	return LegacyMatchList.toEncodedJson(args)
+end
+
+--functions shared between convertMatchMaps and convertSwissMatchMaps
+function LegacyMatchList.copyDetailsToArgs(args, details)
+	for key, value in pairs(details) do
+		if Logic.isEmpty(args[key]) then
+			args[key] = value
+		end
+	end
+	args.details = nil
+	return args
+end
+
+function LegacyMatchList.processMaps(args, details)
+	for index = 1, 15 do
+		if details['map' .. index] then
+			args['map' .. index] = MatchSubobjects.luaGetMap(nil, {
+				map = details['map' .. index],
+				winner = details['map' .. index .. 'win'],
+				score1 = details['map' .. index .. 't1score'],
+				score2 = details['map' .. index .. 't2score'],
+				ot = details['ot' .. index],
+				otlength = details['otlength' .. index],
+				vod = details['vodgame' .. index],
+				comment = details['map' .. index .. 'comment'],
+				})
+			--atm we ignore the old mapXtYgoals parameters, because
+			--1) according to Lukasz they are not used nor disaplayed anymore anyways
+			--2) they are pretty hard to convert, due to the new system wanting goal times
+			--tied to the participants and that info isn't available for the old stuff
+
+			--empty all the stuff we set into this map
+			details['map' .. index] = nil
+			details['map' .. index .. 'win'] = nil
+			details['map' .. index .. 't1score'] = nil
+			details['map' .. index .. 't2score'] = nil
+			details['ot' .. index] = nil
+			details['otlength' .. index] = nil
+			details['vodgame' .. index] = nil
+			details['map' .. index .. 'comment'] = nil
+		else
+			break
+		end
+	end
+	return args, details
+end
+
+--the following function is basically copied from Module:Match
```

Comment:
Seems a little suboptimal

Reply:
yeah it needs some fine tuning, some of isn't used anyways, so i might as well kick some of it
(e.g. the qualwin stuff doesn't fit for LegacyMatchLists)